src/vm/tests/sequences.rs

```diff
@@ -448,6 +448,96 @@ fn test_simple_map_append() {
     );
 }
 
+#[test]
+fn test_slice_list() {
+    let tests = [
+        "(slice (list 2 3 4 5 6 7 8) u0 u3)",
+        "(slice (list u0 u1 u2 u3 u4) u3 u2)",
+        "(slice (list 2 3 4 5 6 7 8) u0 u0)",
+        "(slice (list 2 3 4 5 6 7 8) u10 u3)",
+    ];
+
+    let expected = [
+        Value::some(Value::list_from(vec![Value::Int(2), Value::Int(3), Value::Int(4)]).unwrap()).unwrap(),
+        Value::some(Value::list_from(vec![Value::UInt(3), Value::UInt(4)]).unwrap()).unwrap(),
+        Value::none(),
+        Value::none(),
+    ];
+
+    for (test, expected) in tests.iter().zip(expected.iter()) {
+        assert_eq!(expected.clone(), execute_v2(test).unwrap().unwrap());
+    }
+}
+
+#[test]
+fn test_slice_buff() {
+    let tests = [
+        "(slice 0x000102030405 u0 u3)",
+        "(slice 0x000102030405 u3 u3)",
+        "(slice 0x000102030405 u3 u10)",
+        "(slice 0x000102030405 u10 u3)",
+        "(slice 0x u2 u3)",
+    ];
+
+    let expected = [
+        Value::some(Value::buff_from(vec![0, 1, 2]).unwrap()).unwrap(),
+        Value::some(Value::buff_from(vec![3, 4, 5]).unwrap()).unwrap(),
+        Value::none(),
+        Value::none(),
+        Value::none(),
+    ];
+
+    for (test, expected) in tests.iter().zip(expected.iter()) {
+        assert_eq!(expected.clone(), execute_v2(test).unwrap().unwrap());
+    }
+}
+
+#[test]
+fn test_slice_ascii() {
+    let tests = [
+        "(slice \"blockstack\" u0 u5)",
+        "(slice \"blockstack\" u5 u5)",
+        "(slice \"blockstack\" u5 u0)",
+        "(slice \"blockstack\" u11 u3)",
+        "(slice \"\" u0 u3)",
+    ];
+
+    let expected = [
+        Value::some(Value::string_ascii_from_bytes("block".into()).unwrap()).unwrap(),
+        Value::some(Value::string_ascii_from_bytes("stack".into()).unwrap()).unwrap(),
+        Value::none(),
+        Value::none(),
+        Value::none(),
+    ];
+
+    for (test, expected) in tests.iter().zip(expected.iter()) {
+        assert_eq!(expected.clone(), execute_v2(test).unwrap().unwrap());
+    }
+}
+
+#[test]
+fn test_slice_utf8() {
+    let tests = [
+        "(slice u\"hello \\u{1F98A}\" u0 u5)",
+        "(slice u\"hello \\u{1F98A}\" u6 u1)",
+        "(slice u\"hello \\u{1F98A}\" u6 u0)",
```

Comment:
Are you still returning "None" here? I agree with Reed's comment that it should be empty list.